accepted/2024/support-nuget-authentication-plugins-dotnet-tools.md

```diff
@@ -0,0 +1,504 @@
+# ***Support for NuGet authentication plugins deployed via .NET tools***
+
+- Author Name: <https://github.com/kartheekp-ms>
+- GitHub Issue: <https://github.com/NuGet/Home/issues/12567>
+
+## Summary
+
+Currently, NuGet utilizes a [cross-platform plugin model](https://learn.microsoft.com/nuget/reference/extensibility/nuget-cross-platform-plugins#supported-operations) which is primarily used for [authentication against private feeds](https://learn.microsoft.com/nuget/reference/extensibility/nuget-cross-platform-authentication-plugin).
+It also supports the [package download](https://github.com/NuGet/Home/wiki/NuGet-Package-Download-Plugin) operation.
+
+To accommodate all scenarios involving NuGet client tools, plugin authors will need to create plugins for both `.NET Framework` and `.NET Core`.
+The following details the combinations of client and framework for these plugins.
+
+| Client tool | Framework |
+|-------------|-----------|
+| Visual Studio | .NET Framework |
+| dotnet.exe | .NET Core |
+| NuGet.exe | .NET Framework |
+| MSBuild.exe | .NET Framework |
+| NuGet.exe on Mono | .NET Framework |
+
+Currently, NuGet maintains `netfx` folder for plugins that will be invoked in `.NET Framework` code paths, `netcore` folder for plugins that will be invoked in `.NET Core` code paths.
+
+| Framework | Root discovery location |
+|-----------|------------------------|
+| .NET Core | %UserProfile%/.nuget/plugins/netcore |
+| .NET Framework | %UserProfile%/.nuget/plugins/netfx |
+
+This proposal introduces a new workflow for plugin authors, consumers, and NuGet Client tooling:
+
+- Plugin authors will now be able to publish their NuGet plugins as .NET Tools.
+The only requirement is that the .NET Tool command name should begin with `nuget-plugin-`.
+- Consumers can install these NuGet plugins as [global .NET tools](https://learn.microsoft.com/dotnet/core/tools/global-tools-how-to-use#use-the-tool-as-a-global-tool).
+- Upon installation, these global .NET tools are added to the PATH by the .NET SDK.
+This allows NuGet to easily determine which file in the package should be run at runtime.
```

Comment:
Does this design now address https://github.com/dotnet/sdk/issues/35912?

Reply:
No, it does not. We have suggested a work around https://github.com/dotnet/sdk/issues/35912#issuecomment-2004522180.

Reply:
Wouldn't the path reading method kind of solve that?

Reply:
.NET Tools are installed to a per-user location (`~/.dotnet/tools`), and therefore the PATH is modified by per-user config, not machine config, and therefore `sudo` will still have a different PATH and different `~/.dotnet/tools` location.